nixos/modules/services/admin/pgadmin.nix

```diff
@@ -111,6 +111,8 @@ in
   };
 
   config = mkIf (cfg.enable) {
+    warnings = lib.optional (builtins.any (x: x == pkgs.pgadmin4-desktopmode) config.environment.systemPackages)
+      ''pgadmin4 NixOS module conflicts with pgadmin4-desktopmode, since they share a common config file in /etc/pgadmin'';
```

Comment:
Not sure if this check is necessary/working (it fails to cover scenarios like `nix run`)

Reply:
I agree. I'm open to suggestions. I also added a warning in the description of the package to not use both at the same time. We could add something to the release notes..
This warning at least covers some (most?) cases when using the module and `pgadmin4-desktopmode`

Reply:
A quick grep for `config.environment.systemPackages` shows no prior arts of doing so, and the worst outcome of using them together seems to be a port collision and nothing more?

Reply:
> A quick grep for `config.environment.systemPackages` shows no prior arts of doing so, and the worst outcome of using them together seems to be a port collision and nothing more?

No. Server mode is also set there. Pgadmin4-desktopmode won't start.
I have another idea, but can't test it until tomorrow. I'm thinking of a patch of pgadmin to check for the file in /etc/pgadmin

I'll ping you with an update

Reply:
@NickCao @SuperSandro2000 I added a patch, which will catch a `PermissionError` when `pgadmin4-desktopmode` tries to read the system config. 
With it, both apps get along fine, but it will issue a warning so users are aware of it.